File: algo-c-to-_/src/totient.py
#
#	from src/totient.c
#
#	unsigned phi(unsigned)		to	phi
#

def phi(x):
    t = x
    if x % 2 == 0:
        t, x = t // 2, x // 2
        while x % 2 == 0:
            x = x // 2

    d = 3
    while x // d >= d:
        if x % d == 0:
            t, x = t // d * (d - 1), x // d
            while x % d == 0:
                x = x // d
        d += 2

    if x > 1:
        t = t // x * (x - 1)

    return t

File: porting-from-algo-c/examples/totient.lua
--
--  from src/totient.c
--
--    unsigned phi(unsigned)  to  phi
--

local M = require 'totient'
local H = require '_helper'

local phi = M.phi
local id, tableWriter = H.id, H.tableWriter

do
	tableWriter(
		{ 1, 10, 1 },
		{ 0, 19, 1 },
		{ 4, 4 },
		{ function (n) return n * 10 end, id, function (i, j) return phi(10 * j + i) end },
		{ "d", "d", "d" }
	)()
end
